keyboards/crkbd/keymaps/devdev/keymap.c

```diff
@@ -0,0 +1,429 @@
+    {0, 10, HSV_PINK}
+);
+
+// _SWITCHER   // light up top row
+const rgblight_segment_t PROGMEM layer_switcher_lights[] = RGBLIGHT_LAYER_SEGMENTS( 
+    {0, 6, HSV_GREEN},
+	{9, 2, HSV_GREEN},
+	{17, 2, HSV_GREEN},
+	{23, 2, HSV_GREEN}
+);
+
+
+// Now define the array of layers. Later layers take precedence
+const rgblight_segment_t* const PROGMEM my_rgb_layers[] = RGBLIGHT_LAYERS_LIST(
+    layer_qwerty_lights, 
+	layer_colemakdh_lights,
+	layer_num_lights,// overrides layer 1
+	layer_symbol_lights,
+    layer_command_lights,       
+	layer_numpad_lights, 
+	layer_numpad_rh_lights,
+	layer_move_lights,
+	layer_switcher_lights  // Overrides other layers
+);
+
+void keyboard_post_init_user(void) {
+    // Enable the LED layers
+    rgblight_layers = my_rgb_layers;
+	rgblight_mode(10);// haven't found a way to set this in a more useful way 
+
+}
+
+void matrix_scan_user(void) {
+	#ifdef SSD1306OLED
+		iota_gfx_task();
+    #endif
+}
+
+void matrix_init_user(void) {
+    //#ifdef RGBLIGHT_ENABLE
+    //  RGB_current_mode = rgblight_config.mode;
+    //#endif
+    //SSD1306 OLED init, make sure to add #define SSD1306OLED in config.h
+    #ifdef SSD1306OLED
+        iota_gfx_init(!has_usb());   // turns on the display
+    #endif
+}
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+	//rgblight_setrgb (0x00,  0x00, 0xFF);
+	rgblight_set_layer_state(0, layer_state_cmp(state, _DEFAULTS) && layer_state_cmp(default_layer_state,_QWERTY));
+	rgblight_set_layer_state(1, layer_state_cmp(state, _DEFAULTS) && layer_state_cmp(default_layer_state,_QWERTY));
+	
+
+	rgblight_set_layer_state(2, layer_state_cmp(state, _NUM));
+	rgblight_set_layer_state(3, layer_state_cmp(state, _SYM));
+	rgblight_set_layer_state(4, layer_state_cmp(state, _COMMAND));
+	rgblight_set_layer_state(5, layer_state_cmp(state, _NUMPAD));
+	if (!has_usb())
+		rgblight_set_layer_state(6, layer_state_cmp(state, _NUMPAD));
+	rgblight_set_layer_state(7, layer_state_cmp(state, _MOVE));
+	rgblight_set_layer_state(8, layer_state_cmp(state, _SWITCH));
+    return state;
+}
+
+/*
+bool led_update_user(led_t led_state) {
+    rgblight_set_layer_state(0, led_state.caps_lock);
+    return true;
+}
+*/
+
+
+
+//SSD1306 OLED update loop, make sure to add #define SSD1306OLED in config.h
+#ifdef OLED_DRIVER_ENABLE
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+  if (!is_master) {
+    return OLED_ROTATION_180;  // flips the display 180 degrees if offhand
+  }
+  return rotation;
+}
+// When add source files to SRC in rules.mk, you can use functions.
+const char *read_layer_state(void);
+const char *read_logo(void);
+void set_keylog(uint16_t keycode, keyrecord_t *record);
+const char *read_keylog(void);
+const char *read_keylogs(void);
+
+
+const char *read_mode_icon(bool swap);
+// const char *read_host_led_state(void);
+// void set_timelog(void);
+// const char *read_timelog(void);
+
+
+const char *read_layer_state(void) {
+  switch (get_highest_layer(layer_state))
```

Comment:
This would let you use a single switch to check everything. ```suggestion
  switch (get_highest_layer(layer_state|default_layer_state))
```